src/Cassandra/FutureResponse.php

```diff
@@ -0,0 +1,61 @@
+<?php
+namespace M6Web\Bundle\CassandraBundle\Cassandra;
+
+use Cassandra\Future;
+use M6Web\Bundle\CassandraBundle\EventDispatcher\CassandraEvent;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * Class FutureResponse
+ *
+ * Handle future response for dispatching event when request is complete
+ */
+class FutureResponse implements Future
+{
+    /**
+     * @var EventDispatcher
+     */
+    protected $eventDispatcher;
+
+    /**
+     * @var CassandraEvent
+     */
+    protected $event;
+
+    /**
+     * @var Futurer
+     */
+    protected $future;
+
+    /**
+     * @param Future                   $future
+     * @param CassandraEvent           $event
+     * @param EventDispatcherInterface $eventDispatcher
+     */
+    public function __construct(Future $future, CassandraEvent $event, EventDispatcherInterface $eventDispatcher)
+    {
+        $this->future          = $future;
+        $this->event           = $event;
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    /**
+     * Waits for a given future resource to resolve and throws errors if any.
+     *
+     * @param int|null $timeout
+     *
+     * @throws \Cassandra\Exception\InvalidArgumentException
+     * @throws \Cassandra\Exception\TimeoutException
+     *
+     * @return mixed a value that the future has been resolved with
+     */
+    public function get($timeout = null)
+    {
+        $return = $this->future->get($timeout);
+
+        $this->event->setExecutionStop();
```

Comment:
In this case, the computed executionTime is not the "real" execution time. The request can have terminated before the client call the `get()` on the `Future`.

Unfortunately, I don't see a way to get the "real" value, so we can probably keep it as is. We should maybe add a warning about this in the (future) documentation.